### PR TITLE
(maint) Cover logs generated by ensure file type

### PR DESCRIPTION
**Pull request by MikaelSmith**
Without this commit, the File type has no tests that verify the logs
generated when the state is changed.

Add tests that verify the logs generated when
- creating a new file
- overwriting an existing file
- removing a file

### CODE REVIEW

**Comment by MikaelSmith** [comment]
Tests for the File type, related to changes proposed in #3735.

**Comment by branan** [comment]
Still waiting on travis

File: spec/integration/type/file_spec.rb [hunk 1]
```diff
@@ -94,19 +94,39 @@ def get_aces_for_path_by_sid(path, sid)
...
     end
 
-    it "should do nothing if file is not present" do
-      catalog.add_resource(described_class.new(:path => path, :ensure => :absent, :backup => :false))
-      report = catalog.apply.report
-      expect(report.resource_statuses["File[#{path}]"]).not_to be_failed
-      expect(Puppet::FileSystem.exist?(path)).to be_falsey
+    context "file is present" do
+      before(:each) do
+        FileUtils.touch(path)
```
Comment (branan) [inline spec/integration/type/file_spec.rb]: Are we using some sort of filesystem mocking library? This looks not particularly unit-y if we're not

**Comment by MikaelSmith** [comment]
Travis should be a simple fix, assumed all file creation would use the same default permissions.

File: spec/integration/type/file_spec.rb (hunk 1 above)
Comment (MikaelSmith) [inline spec/integration/type/file_spec.rb]: These are spec/integration, not spec/unit. I don't think they generally mock the filesystem.

**Comment by branan** [comment]
:+1: these tests look reasonable (or as reasonable as trying to validate logging ever is).

**Comment by MikaelSmith** [comment]
Rebased to pick up PUP-4231 changes, just in case.

File: spec/integration/type/file_spec.rb [hunk 2]
```diff
@@ -356,14 +376,26 @@ def get_aces_for_path_by_sid(path, sid)
...
+              before :each do
+                FileUtils.touch(path)
+                File.chmod(0644, path)
+                catalog.add_resource described_class.new(:path => path, :source => link, :mode => '0600', :links => :follow)
+              end
 
-              catalog.add_resource described_class.new(:path => path, :source => link, :mode => '0600', :links => :follow)
-              catalog.apply
+              it "should overwrite the file" do
+                report = catalog.apply.report
```
Comment (elyscape) [inline spec/integration/type/file_spec.rb]: `report` isn't used in this test.

**Pull request merged**